File: app/src/main/java/com/example/inklink/models/ReportedUsers.java
package com.example.inklink.models;

public class ReportedUsers {
    private int id, userId, reporterId;
    private String reportType, reportStatus;

    public ReportedUsers () {
        this.reportStatus = "unhandled";
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUserId() {
        return userId;
    }

    public void setUserId(int userId) {
        this.userId = userId;
    }

    public void setReporterId(int reporterId) { this.reporterId = reporterId; }

    public int getReporterId() { return reporterId; }

    public String getReportType() {
        return reportType;
    }

    public void setReportType(String reportType) {
        this.reportType = reportType;
    }

    public String getReportStatus() {
        return reportStatus;
    }

    public void setReportStatus(String reportStatus) {
        this.reportStatus = reportStatus;
    }
}

File: app/src/main/java/com/example/inklink/models/ReportedUsers.kt
package com.example.inklink.models

/**
 * This class is a model for 'reported_users' table and contains all the columns of it.
 */
class ReportedUsers {
    var id: Int = 0
    var userId: Int = 0
    var reporterId: Int = 0
    var reportType: String? = null
    var reportStatus: String? = null

    init {
        this.reportStatus = "unhandled"
    }

    override fun toString(): String {
        return """
            ReportedUser: {
                id: ${this.id},
                userId: ${this.userId},
                reporterId: ${this.reporterId},
                reportType: ${this.reportType},
                reportStatus: ${this.reportStatus},
            }
        """.trimIndent()
    }
}
